nilearn/glm/tests/test_first_level.py

```diff
@@ -1860,3 +1860,30 @@ def test_missing_trial_type_column_warning(tmp_path_factory):
             "No column named 'trial_type' found" in r.message.args[0]
             for r in record
         )
+
+
+def test_first_level_from_bids_load_confounds(tmp_path):
+    """Test several BIDS structure."""
+    n_sub = 2
+
+    bids_path = create_fake_bids_dataset(
+        base_dir=tmp_path, n_sub=n_sub, n_ses=2, tasks=["main"], n_runs=[2]
+    )
+
+    models, m_imgs, m_events, m_confounds = first_level_from_bids(
+        dataset_path=bids_path,
+        task_label="main",
+        space_label="MNI",
+        img_filters=[("desc", "preproc")],
+        slice_time_ref=None,
+        confounds_strategy=("motion", "wm_csf", "scrub"),
+        confounds_motion="full",
+        confounds_wm_csf="basic",
+        confounds_scrub=1,
+        confounds_fd_threshold=0.2,
+        confounds_std_dvars_threshold=3,
+    )
```

Comment:
@htwangtw @ymzayek @bthirion 
keeping this as a draft so you can discuss API, implementation...

Will update the doc once the dust settles.

Reply:
I guess this could be made lighter if we expect to have predefined configurations for confounds, but I don't think that we're at that point.
At least I find the current API quite explicit.

Reply:
Actually, we might get rid of `confounds_strategy`, because the next three arguments are redundant with the provided list ?

Reply:
Note this test just show a subset of the possible arguments that can be passed to load_confounds

> I guess this could be made lighter if we expect to have predefined configurations for confounds, but I don't think that we're at that point.

> Actually, we might get rid of `confounds_strategy`, because the next three arguments are redundant with the provided list ?

Actually I am just reusing the API from load_confounds, so it can almost be passed as is and we can let load_confounds do the argument validation

https://nilearn.github.io/dev/modules/generated/nilearn.interfaces.fmriprep.load_confounds.html

In short `strategy` defines what type of confounds to include and all the other parameters give more details on how to include them.

Reply:
will update the doc string to try to explain this so we can see if it makes sense